components/NavBar.tsx

```diff
@@ -0,0 +1,48 @@
+import { AppBar, Toolbar, Typography, Button, makeStyles } from '@material-ui/core'
+import React, { useContext } from 'react'
+import NextLink from 'next/link'
+
+import ElevationScroll from './ElevationScroll'
+import { AppContext } from './AppContext'
+
+interface Props {
+	title: string
+}
+
+const useStyles = makeStyles({
+	title: {
+		flexGrow: 1,
+	},
+})
+
+export default function NavBar(props: Props): JSX.Element {
+	const { setThemeType, themeType } = useContext(AppContext)
+	const classes = useStyles()
+
+	return (
+		<ElevationScroll>
+			<AppBar position="sticky">
+				<Toolbar>
+					<NextLink href="/">
+						<Typography variant="h5" className={classes.title}>
+							{props.title}
+						</Typography>
+					</NextLink>
+					<NextLink href="/about">
+						<Button color="inherit">About</Button>
+					</NextLink>
+					<Button
+						color="inherit"
+						onClick={(): void => {
+							if (setThemeType) {
+								setThemeType(themeType === 'dark' ? 'light' : 'dark')
+							}
+						}}
+					>
+						Invert Theme
+					</Button>
```

Comment:
the theme toggle button should probably be its own component. Bonus points if it can be used with mui's routing integration e.g. `<Button component={ThemeToggleLink}>`